sklearn/metrics/cluster/unsupervised.py

```diff
@@ -81,6 +82,8 @@ def silhouette_score(X, labels, metric='euclidean', sample_size=None,
            <http://en.wikipedia.org/wiki/Silhouette_(clustering)>`_
 
     """
+    if np.array(X).ndim == 1:
```

Comment:
You should do `X = np.asarray(X)` here and check to avoid another array conversion inside `check_X_y` if X is a list etc